frontends/concrete-python/concrete/fhe/mlir/converter.py

```diff
@@ -500,12 +502,17 @@ def round_bit_pattern(self, ctx: Context, node: Node, preds: List[Conversion]) -
                 shifter //= 2
 
             if shifter != 1:
-                pred = ctx.mul(pred.type, pred, ctx.constant(ctx.i(pred.bit_width + 1), shifter))
+                shift_amount = int(np.log2(shifter))
```

Comment:
could be the helper function cast_to with optional shift_amount

Reply:
I think this is fine for now, it's only used in one place.